tests/functional_tests/run_func_test.py

```diff
@@ -539,6 +540,238 @@ def test_out_id_only(self):
+                self.generate_command()
+                os.system(self.command)
+
+                self.output_file_path = os.path.join(
+                    self.output_path,
+                    f"{self.output_file_prefix}.{self.output_file_to_check}.txt",
+                )
+                self.expected_file_path = os.path.join(
+                    self.path, f"true-{self.output_file_to_check}-{self.test_cases}.txt"
+                )
+
+                self.output = read_and_sort_file(self.output_file_path)
+                self.expected = read_and_sort_file(self.expected_file_path)
+
+                assert self.output == self.expected
+
+                self.input_files.pop(-1)
+                self.input_file_depend_on_test_cases.pop(-1)
+
+            elif self.test_cases == "est_alt_allele_prob_single":
+                self.arguments["est_alt_allele_prob"] = None
+                self.output_file_to_check = "alt_allele_prob"
+
+                self.generate_command()
+                os.system(self.command)
+
+                self.output_file_path = os.path.join(
+                    self.output_path,
+                    f"{self.output_file_prefix}.{self.output_file_to_check}.txt",
+                )
+                self.expected_file_path = os.path.join(
+                    self.path, f"true-{self.output_file_to_check}-{self.test_cases}.txt"
+                )
+
+                self.output = read_and_sort_file(self.output_file_path)
+                self.expected = read_and_sort_file(self.expected_file_path)
+
+                assert self.output == self.expected
+
+            elif self.test_cases == "est_alt_allele_prob_multiple":
+                self.generate_command()
+                os.system(self.command)
+
+                self.output_file_path = os.path.join(
+                    self.output_path,
+                    f"{self.output_file_prefix}.{self.output_file_to_check}.txt",
+                )
+                self.expected_file_path = os.path.join(
+                    self.path, f"true-{self.output_file_to_check}-{self.test_cases}.txt"
+                )
+
+                self.output = read_and_sort_file(self.output_file_path)
+                self.expected = read_and_sort_file(self.expected_file_path)
+
+                assert self.output == self.expected
+
+            elif self.test_cases == "both":
+                self.input_files.append("alt_allele_prob_file")
+                self.input_file_depend_on_test_cases("alt_allele_prob_file")
+
+                self.generate_command()
+                os.system(self.command)
+
+                self.output_file_path = os.path.join(
+                    self.output_path,
+                    f"{self.output_file_prefix}.{self.output_file_to_check}.txt",
+                )
+
+                self.output = read_and_sort_file(self.output_file_path)
+
+                # check if the estimated alt_allele_prob is in between 0.5 and 1 (include 0.5 and exclude 1)
```

Comment:
@XingerTang I have not looked at the input files, but is this what we expect for the input files - that alt allele prob will be [0.5, 1]?

Reply:
@gregorgorjanc It is specific to the test case and the test input data I designed. In general, the input values for alt allele prob should in [0, 1].